src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +151,2291 @@ static ColliderQuadInit D_80B4A204 = {
+    Collider_InitCylinder(globalCtx, &this->bodyCollider);
+    Collider_SetCylinder(globalCtx, &this->bodyCollider, thisx, &sBodyCylinderInit);
+    Collider_InitQuad(globalCtx, &this->swordCollider);
+    Collider_SetQuad(globalCtx, &this->swordCollider, thisx, &sSwordQuadInit);
+
+    if (thisx->params == ENZF_TYPE_DINOLFOS) { // Dinalfos
+        thisx->colChkInfo.health = 12;
+        thisx->naviEnemyId = 0x10;
+        SkelAnime_Init(globalCtx, &this->skelAnime, &gZfDinolfosSkel, &gZFCryingAnim, this->jointTable,
+                       this->morphTable, ENZF_LIMB_MAX);
+    } else { // Lizalfos
+        thisx->colChkInfo.health = 6;
+        thisx->naviEnemyId = 0xF;
+        SkelAnime_Init(globalCtx, &this->skelAnime, &gZfLizalfosSkel, &gZFCryingAnim, this->jointTable,
+                       this->morphTable, ENZF_LIMB_MAX);
+    }
+
+    if (thisx->params < ENZF_TYPE_LIZALFOS_MINIBOSS_A) { // Not minibosses
+        this->homePlatform = this->curPlatform = -1;
+        D_80B4A1B4 = -1;
+        this->hopAnimIndex = 1;
+        if (thisx->params == ENZF_TYPE_LIZALFOS_LONE) {
+            EnZf_SetupDropIn(this);
+        } else { // Dinolfos
+            func_80B45384(this);
+        }
+    } else { // Minibosses
+        posDiff = player->actor.world.pos.y - thisx->world.pos.y;
+
+        if ((ABS(posDiff) <= 100.0f) && !Flags_GetSwitch(globalCtx, this->clearFlag)) {
+            this->homePlatform = this->curPlatform = EnZf_FindPlatform(&thisx->world.pos, 0);
+            EnZf_SetupDropIn(this);
+            D_80B4A1B4 = 1;
+        } else {
+            Actor_Kill(thisx);
+        }
+    }
+}
+
+void EnZf_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnZf* this = THIS;
+
+    // miniboss
+    if ((this->actor.params >= ENZF_TYPE_LIZALFOS_MINIBOSS_A) &&
+        (Actor_FindNearby(globalCtx, &this->actor, ACTOR_EN_ZF, ACTORCAT_ENEMY, 10000.0f) == NULL)) {
+        func_800F5B58();
+    }
+
+    Effect_Delete(globalCtx, this->blureIndex);
+    Collider_DestroyCylinder(globalCtx, &this->bodyCollider);
+    Collider_DestroyQuad(globalCtx, &this->swordCollider);
+}
+
+/**
+ * Finds the index of the platform position sufficiently close to pos, starting from startIndex and working down.
+ */
+s16 EnZf_FindPlatform(Vec3f* pos, s16 startIndex) {
+    f32 rangeXZ;
+    s16 i;
+
+    rangeXZ = 210.0f;
+
+    // Upstairs has a smaller range
+    if (pos->y >= 420.0f) {
+        rangeXZ = 110.0f;
+    }
+
+    if ((startIndex != -1) && ((sPlatformPositions[startIndex].y - 150.0f) <= pos->y) &&
+        (pos->y <= (sPlatformPositions[startIndex].y + 150.0f)) &&
+        ((sPlatformPositions[startIndex].x - rangeXZ) <= pos->x) &&
+        (pos->x <= (sPlatformPositions[startIndex].x + rangeXZ)) &&
+        ((sPlatformPositions[startIndex].z - rangeXZ) <= pos->z) &&
+        (pos->z <= (sPlatformPositions[startIndex].z + rangeXZ))) {
+        return startIndex;
+    }
+
+    for (i = ARRAY_COUNT(sPlatformPositions) - 1; i > -1; i--) {
```

Comment:
```suggestion
    for (i = ARRAY_COUNT(sPlatformPositions) - 1; i >= 0; i--) {
```

Reply:
Given that `-1` is used as a sort of "default" in this function, I think it's better to keep it (Slightly tempting to use a define, even...)